WalletWasabi/WabiSabi/Backend/Rounds/Arena.Partial.cs

```diff
@@ -67,19 +67,21 @@ private async Task<InputRegistrationResponse> RegisterInputCoreAsync(InputRegist
 			// only that the probability of duplicates is very low).
 			var id = new Guid(SecureRandom.Instance.GetBytes(16));
 
-			var isPayingZeroCoordinationFee = CoinJoinIdStore.Contains(coin.Outpoint.Hash);
+			var comingFromCoinJoin = CoinJoinIdStore.Contains(coin.Outpoint.Hash);
+			bool oneHop = false;
 
-			if (!isPayingZeroCoordinationFee)
+			if (!comingFromCoinJoin)
 			{
 				// If the coin comes from a tx that all of the tx inputs are coming from a CJ (1 hop - no pay).
 				Transaction tx = await Rpc.GetRawTransactionAsync(coin.Outpoint.Hash, true, cancellationToken).ConfigureAwait(false);
 
 				if (tx.Inputs.All(input => CoinJoinIdStore.Contains(input.PrevOut.Hash)))
 				{
-					isPayingZeroCoordinationFee = true;
+					oneHop = true;
 				}
 			}
 
+			var isPayingZeroCoordinationFee = comingFromCoinJoin && oneHop;
```

Comment:
If we are coming from a CJ, the `oneHop` stays false and we pay coord fee.

```suggestion
			var isPayingZeroCoordinationFee = comingFromCoinJoin || oneHop;
```